### ISSUE TITLE
[Bug] When there are too many orders, it is impossible to test Order Email

### ISSUE THREAD

**Issue by Roshyo** [synolia/SyliusMailTesterPlugin#26]
**Describe the bug**
Due to the fact that this plugin is using `EntityType`, if you have too many orders, then the page for testing Order email just won't load

**To Reproduce**
Have a lot of orders
1. Go to Order mail test page
2. Wait for timeout

**Expected behavior**
We should use a ResourceAutocomplete type for relations

**Comment by igormukhingmailcom** [comment]
Can be fixed by adding and using something like `LimitedEntityType` instead of `EntityType`:

```php
<?php

declare(strict_types=1);

namespace App\Mailer\MailTester\Form\Type;

use Doctrine\ORM\QueryBuilder;
use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
use Symfony\Bridge\Doctrine\Form\Type\EntityType;
use Symfony\Component\Form\Exception\UnexpectedTypeException;
use Symfony\Component\OptionsResolver\Options;
use Symfony\Component\OptionsResolver\OptionsResolver;

class LimitedEntityType extends EntityType
{
    public function configureOptions(OptionsResolver $resolver)
    {
        parent::configureOptions($resolver);

        $queryBuilderNormalizer = function (Options $options, $queryBuilder) {
            if (\is_callable($queryBuilder)) {
                $queryBuilder = $queryBuilder($options['em']->getRepository($options['class']));

                if (null !== $queryBuilder && !$queryBuilder instanceof QueryBuilder) {
                    throw new UnexpectedTypeException($queryBuilder, 'Doctrine\ORM\QueryBuilder');
                }
            } else {
                /** @var EntityRepository $repository */
                $repository = $options['em']->getRepository($options['class']);
                $queryBuilder = $repository->createQueryBuilder('o');
            }

            $queryBuilder->setMaxResults($options['limit']);

            return $queryBuilder;
        };

        $resolver
            ->setNormalizer('query_builder', $queryBuilderNormalizer)

            ->setDefined('limit')
            ->setDefault('limit', 20)
            ->setAllowedTypes('limit', 'int')
        ;
    }
}
```

**Comment by Roshyo** [comment]
But wouldn't this prevent from using any order ? I mean, it is not a search field, is it ?

**Comment by igormukhingmailcom** [comment]
Yes, it just limiting entities. Something like ResourceAutocompleteType will be much better.

**Comment by oallain** [comment]
Fixed in [v2.2.0](https://github.com/synolia/SyliusMailTesterPlugin/releases/tag/v2.2.0).

Thanks all !
